### PR TITLE
`FlutterMouse.*` -> `kFlutterMouse.*`, so we can lint header files.

### PR DESCRIPTION
**Pull request by matanlurey**
Partial work towards https://github.com/flutter/flutter/issues/134969.

### CODE REVIEW

**Review by chunhtai — APPROVED** [review]
LGTM, though are these public API? cc @cbracken

**Review by cbracken — CHANGES_REQUESTED** [review]
The change to the public framework header is a breaking change. Please hold off on landing this one.

File: shell/platform/darwin/macos/framework/Headers/FlutterViewController.h [hunk 1]
```diff
@@ -14,13 +14,13 @@
  */
 typedef NS_ENUM(NSInteger, FlutterMouseTrackingMode) {
   // Hover events will never be sent to Flutter.
-  FlutterMouseTrackingModeNone = 0,
+  kFlutterMouseTrackingModeNone = 0,
```
Comment (cbracken) [inline shell/platform/darwin/macos/framework/Headers/FlutterViewController.h:17]:
```suggestion
  kFlutterMouseTrackingModeNone = 0,
  FlutterMouseTrackingModeNone __attribute__((deprecated)) = kFlutterMouseTrackingModeNone,
```

File: shell/platform/darwin/macos/framework/Headers/FlutterViewController.h [hunk 2]
```diff
@@ -14,13 +14,13 @@
...
   // Hover events will be sent to Flutter when the view is in the key window.
-  FlutterMouseTrackingModeInKeyWindow,
+  kFlutterMouseTrackingModeInKeyWindow,
```
Comment (cbracken) [inline shell/platform/darwin/macos/framework/Headers/FlutterViewController.h:19]:
```suggestion
  kFlutterMouseTrackingModeInKeyWindow = 1,
  FlutterMouseTrackingModeInKeyWindow __attribute__((deprecated)) = kFlutterMouseTrackingModeInKeyWindow,
```

File: shell/platform/darwin/macos/framework/Headers/FlutterViewController.h [hunk 3]
```diff
@@ -14,13 +14,13 @@
...
   // Hover events will be sent to Flutter when the view is in the active app.
-  FlutterMouseTrackingModeInActiveApp,
+  kFlutterMouseTrackingModeInActiveApp,
```
Comment (cbracken) [inline shell/platform/darwin/macos/framework/Headers/FlutterViewController.h:21]:
```suggestion
  kFlutterMouseTrackingModeInActiveApp = 2,
  FlutterMouseTrackingModeInActiveApp __attribute__((deprecated)) = kFlutterMouseTrackingModeInActiveApp,
```

File: shell/platform/darwin/macos/framework/Headers/FlutterViewController.h [hunk 4]
```diff
@@ -14,13 +14,13 @@
...
   // Hover events will be sent to Flutter regardless of window and app focus.
-  FlutterMouseTrackingModeAlways,
+  kFlutterMouseTrackingModeAlways,
```
Comment (cbracken) [inline shell/platform/darwin/macos/framework/Headers/FlutterViewController.h:23]:
```suggestion
  kFlutterMouseTrackingModeAlways = 3,
  FlutterMouseTrackingModeAlways __attribute__((deprecated)) = kFlutterMouseTrackingModeAlways,
```

**Pull request merged**